contracts/Market.sol

```diff
@@ -57,9 +57,72 @@ contract Market is PausableAccessPreset {
 
   /**
    * @notice Emitted on setting of `_priorityRestrictedThreshold`.
+   * @param threshold The updated threshold for priority restricted supply.
    */
   event PriorityRestrictedThresholdSet(uint256 threshold);
 
+  /**
+   * @notice Emitted on setting of `_noriFeeWalletAddress`.
+   * @param updatedWalletAddress The updated address of the Nori fee wallet.
+   */
+  event NoriFeeWalletAddressUpdated(address updatedWalletAddress);
+
+  /**
+   * @notice Emitted on setting of `_noriFeePercentage`.
+   * @param updatedFeePercentage The updated fee percentage for Nori.
+   */
+  event NoriFeePercentageUpdated(uint256 updatedFeePercentage);
+
+  /**
+   * @notice Emitted on setting of `_currentSupplierAddress`.
+   * @param updatedSupplierAddress The updated supplier address.
+   */
+  event CurrentSupplierAddressChanged(address updatedSupplierAddress);
+
+  /**
+   * @notice Emitted when adding a supplier to `_activeSupply`.
+   * @param addedSupplierAddress The supplier that was added.
+   * @param nextSupplierAddress The next of the supplier that was added, updated to point to `addedSupplierAddress` as previous.
+   * @param previousSupplierAddress The previous of the supplier that was added, updated to point to `addedSupplierAddress` as next.
+   */
+  event SupplierAdded(
+    address indexed addedSupplierAddress,
+    address indexed nextSupplierAddress,
+    address indexed previousSupplierAddress
+  );
+
+  /**
+   * @notice Emitted when removing a supplier from `_activeSupply`.
+   * @param removedSupplierAddress The supplier that was removed.
+   * @param nextSupplierAddress The next of the supplier that was removed, updated to point to `previousSupplierAddress` as previous.
+   * @param previousSupplierAddress The previous of the supplier that was removed, updated to point to `nextSupplierAddress` as next.
+   */
+  event SupplierRemoved(
+    address indexed removedSupplierAddress,
+    address indexed nextSupplierAddress,
+    address indexed previousSupplierAddress
+  );
+
+  /**
+   * @notice Emitted when a removal is added to `_activeSupply`.
+   * @param removalId The removal that was added.
+   * @param supplierAddress The address of the supplier for the removal.
+   */
+  event RemovalAdded(
+    uint256 indexed removalId,
+    address indexed supplierAddress
+  );
```

Comment:
the changes I'm making in [this PR ](https://github.com/nori-dot-eco/contracts/pull/337) change it so that removals can be added to a `RemovalQueue` without reverting even if they already existed in the queue.  I wonder if we should either push this event emission into the `RemovalQueue` library and predicate it on the return value from `enumerableset.add(removalId)`, or if we should have `insertRemovalByVintage` return true or false based on whether the set had already contained the removal, and then predicate the emission of this event on that return value.  Just so that we're only emitting this event when a removal was truly added to the queue